vault/token_store.go

```diff
@@ -3502,6 +3523,8 @@ func (ts *TokenStore) authRenew(ctx context.Context, req *logical.Request, d *fr
 
 	req.Auth.Period = role.TokenPeriod
 	req.Auth.ExplicitMaxTTL = role.TokenExplicitMaxTTL
+	req.Auth.TTL = role.TokenTTL
```

Comment:
Suppose I have a token that was created by a role prior to this change, then I upgrade to a newer Vault version and do a renew.  Won't this zero out the token's TTL, leading to a change in renew behaviour?